test/absinthe/phase/document/validation/arguments_of_correct_type_test.exs

```diff
@@ -974,6 +974,36 @@ defmodule Absinthe.Phase.Document.Validation.ArgumentsOfCorrectTypeTest do
     end
   end
 
+  describe "Invalid Custom Scalar value" do
+    test "Invalid scalar input on mutation, no suggestion" do
+      assert_fails_validation(
+        """
+        mutation($scalarInput: CustomScalar) {
+          createDog(customScalarInput: $scalarInput)
+        }
+        """,
+        [
+          variables: %{
+            "scalarInput" => [
+              %{
+                "foo" => "BAR"
```

Comment:
Ah I think I see the situation here. When you supply a map instead of a scalar, these inner keys get turned into input object input "fields", but then there isn't any corresponding schema node to match to, since it's supposed to be a scalar. Is that the idea?